src/storage-types/src/controller.rs

```diff
@@ -441,7 +442,7 @@ impl TxnsCodec for TxnsCodecRow {
         if batch.is_null() {
             TxnsEntry::Register(data_id, ts)
         } else {
-            TxnsEntry::Append(data_id, ts, batch.unwrap_bytes().to_vec())
+            TxnsEntry::Append(data_id, ts, Bytes::copy_from_slice(batch.unwrap_bytes()))
```

Comment:
It's a shame that we copy from slice here because we end up dropping `SourceData` and the underlying byte array. It would be much more satisfying if we could move out of `SourceData` instead of copying.